### PR TITLE
Exploratory Data Analysis (EDA)

### PR DESCRIPTION
**Pull request by majercakdavid**
## Purpose
Demo for demonstration of how one can perform exploratory data analysis in FL

## Does this introduce a breaking change?

```
[ ] Yes
[ x] No
```

## Pull Request Type
Addition of new demo component

```
[ ] Bugfix
[x ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What to Check
Output of `data_analysis` component
* ...

### CODE REVIEW

**Comment by majercakdavid** [comment]
Example run: https://ml.azure.com/runs/helpful_peach_y1v89rcfjv?wsid=/subscriptions/48bbc269-ce89-4f6f-9a12-c6f91fcb772d/resourcegroups/fldev-customer1-rg/workspaces/aml-fldemo-customer1&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

File: examples/pipelines/ccfraud/config.yaml [hunk 1]
```diff
@@ -8,6 +8,8 @@ aml:
 
 # federated learning parameters
 federated_learning:
+  run_data_analysis: true
```
Comment (jfomhover) [inline examples/pipelines/ccfraud/config.yaml:11]: could you move this parameter under either data_analysis_parameters or training_parameters ?

federated_learning is really a section that is only about configuring the silos and orchestrator.

Comment (jfomhover) [reply examples/pipelines/ccfraud/config.yaml:11]: also, can we turn it off by default? (I think we need to document it better if we want to turn it on by default)

File: examples/components/utils/data_analysis/run.py [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+    }
+    mlflow.log_dict(
+        column_description, f"{metrics_prefix}/{column_name}/column_description.json"
+    )
+
+    fig, ax = plt.subplots()
+    ax = df[column_name].plot(
+        kind="density", xlim=(df[column_name].min(), df[column_name].max())
+    )
+    mlflow.log_figure(fig, f"{metrics_prefix}/{column_name}/density.png")
```
Comment (jfomhover) [inline examples/components/utils/data_analysis/run.py:101]: can we record all those mlflow calls to the pipeline root level?

This is what is done in the training script, and avoids having users to navigate to each step to find the relevant metrics

File: examples/components/utils/data_analysis/run.py [hunk 3]
```diff
@@ -0,0 +1,240 @@
+import argparse
```
Comment (jfomhover) [inline examples/components/utils/data_analysis/run.py:1]: could we rename this component tabular_data_analysis ? (we might have a text data analysis later?)

**Review by jfomhover — CHANGES_REQUESTED** [review]
For a future PR, I'd love for this data analysis tool to really "gather" something to a central location, so we could plot the data distributions in one single plot.

But the current design is ok. There's a couple comments though about params and others that I'd love to take into account before merging. Thanks!

File: docs/provisioning/external-silos.md [hunk 4]
```diff
@@ -94,8 +94,6 @@ az k8s-extension create --name <extension-name> --extension-type Microsoft.Azure
 ```
 where `<extension-name>` can be chosen arbitrarily.
 
-> Note: if you're using an AKS cluster (as opposed to a local k8s cluster), you'll need to change the `--cluster-type` parameter value from `connectedClusters` to `managedClusters`.
```
Comment (jfomhover) [inline docs/provisioning/external-silos.md:97]: is this intentional here?

**Comment by majercakdavid** [comment]
https://ml.azure.com/runs/sleepy_steelpan_pzpp1l5tqg?wsid=/subscriptions/48bbc269-ce89-4f6f-9a12-c6f91fcb772d/resourcegroups/fldev-damajrecak-rg/workspaces/aml-fldevdmdemo0&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

File: examples/components/utils/data_analysis/run.py (hunk 3 above)
Comment (majercakdavid) [reply examples/components/utils/data_analysis/run.py:1]: Could we make this change later on when we decide to add additional types of analyses? Maybe we can make it part of this component by just extending it - I believe the code for analysis itself won't be long.

**Review by jfomhover — APPROVED** [review]

**Pull request merged**